bulk-model-sync-lib/src/jvmTest/kotlin/org/modelix/model/sync/bulk/ModelImporterOrderPropertyTest.kt

```diff
@@ -0,0 +1,508 @@
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 moved with because of the role change from "uid2" to "uid3".
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
+        // child1 changes its role from "uid2" to "uid3".
+        assertEquals("uid3", children[1].getContainmentLink()!!.getUID())
+    }
+
+    @Test
+    fun `unordered children with changing parent in import are appended`() {
+        // language=json
+        val dataBeforeImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              },
+              {
+                "id": "parent2",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataBeforeImport = ModelData.fromJson(dataBeforeImportJson)
+        val branch = loadModelIntoBranch(dataBeforeImport)
+        val modelImporter = ModelImporter(branch.getRootNode())
+
+        // language=json
+        val dataToImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2,"
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 is added after existing child2 because it was moved from another node
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
+    }
+
+    @Test
+    fun `unordered child can not change concept`() {
```

Comment:
```suggestion
    fun `an unordered child cannot change concept`() {
```

Why is this actually the case?

Reply:
This is the case because it is generally not intended for nodes in MPS and our `INode`s to change concepts.
So in the importer, we cannot change the concept.
If we are requested to but cannot change the concept, we fail not to produce garbage data.
(Though this check is not done not properly in all existing path, but this is a change for some other PR.) 

---

I removed this test becaus it is not specific to ordering anymore.